docs/aws/audit/esmonitoring/rules/es_public_service_domain.mdx

```diff
@@ -23,6 +23,247 @@ HIPAA, PCIDSS, NIST
+   aws ec2 authorize-security-group-ingress --group-id <sg-id> --protocol tcp --port 443 --cidr 0.0.0.0/0
+   ```
+
+3. **Create the ElasticSearch Domain with VPC Configuration:**
+   Use the `create-elasticsearch-domain` command to create the domain with VPC configuration. Ensure you specify the VPC ID, subnet IDs, and security group IDs.
+
+   ```sh
+   aws es create-elasticsearch-domain --domain-name my-es-domain --elasticsearch-version 7.10 --vpc-options SubnetIds=<subnet-id>,SecurityGroupIds=<sg-id>
+   ```
+
+4. **Verify the Configuration:**
+   Verify that the ElasticSearch domain is created with the VPC configuration by describing the domain.
+
+   ```sh
+   aws es describe-elasticsearch-domain --domain-name my-es-domain
+   ```
+
+By following these steps, you ensure that your ElasticSearch domain is launched with private VPC endpoints, enhancing security and preventing public access.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent ElasticSearch Domains from being launched without private VPC endpoints using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to ensure that your ElasticSearch domains are launched with private VPC endpoints:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip:
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Initialize a Boto3 session with the necessary AWS credentials and region.
+
+3. **Check Existing Domains**:
+   List all existing ElasticSearch domains and check their VPC configuration.
+
+4. **Create/Update Domain with VPC Configuration**:
+   Ensure that any new or existing ElasticSearch domains are created or updated with the appropriate VPC configuration.
+
+Here is a Python script that demonstrates these steps:
+
+```python
+import boto3
+
+# Initialize a Boto3 session
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='YOUR_REGION'
+)
+
+# Create an Elasticsearch client
+es_client = session.client('es')
+
+# Function to check and ensure VPC configuration
+def ensure_vpc_endpoint(domain_name, vpc_options):
+    try:
+        # Describe the domain to check its current configuration
+        response = es_client.describe_elasticsearch_domain(DomainName=domain_name)
+        domain_status = response['DomainStatus']
+        
+        # Check if the domain is already in a VPC
+        if 'VPCOptions' in domain_status:
+            print(f"Domain {domain_name} is already in a VPC.")
+        else:
+            print(f"Domain {domain_name} is not in a VPC. Updating configuration...")
+            # Update the domain to include VPC options
+            es_client.update_elasticsearch_domain_config(
+                DomainName=domain_name,
+                VPCOptions=vpc_options
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>